src/track/trackmetadatataglib.cpp

```diff
@@ -967,6 +1032,20 @@ void readTrackMetadataFromID3v2Tag(TrackMetadata* pTrackMetadata,
         pTrackMetadata->setAlbumArtist(toQStringFirstNotEmpty(albumArtistFrame));
     }
 
+	QString rating;
+    const TagLib::ID3v2::FrameList popmRatingFrameList(tag.frameListMap()["POPM"]);
+	if (!popmRatingFrameList.isEmpty()) {
+		rating = QString::number(dynamic_cast<TagLib::ID3v2::PopularimeterFrame*>(popmRatingFrameList.front())->rating());
+		for (TagLib::ID3v2::FrameList::ConstIterator it(popmRatingFrameList.begin());
+			it != popmRatingFrameList.end(); it++){
+		        auto popmFrame = dynamic_cast<TagLib::ID3v2::PopularimeterFrame*>(*it);
```

Comment:
The result of a dynamic_cast should always be checked! See various examples in this file